src/colorchannels.jl

```diff
@@ -186,6 +186,27 @@ extractchannels(c::TransparentGray) = (gray(c), alpha(c))
 extractchannels(c::Color3)          = (comp1(c), comp2(c), comp3(c))
 extractchannels(c::Transparent3)    = (comp1(c), comp2(c), comp3(c), alpha(c))
 
+
+
+function _channelview(ex::Expr)
+    args = [ex.args[i] for i = 1:length(ex.args)]
+    for j = 1:length(args)
+        if isa(eval(ex.args[j]), AbstractArray)
```

Comment:
CI errors because `eval(:x)` is executed in the Main scope, this sounds like a common macro issue. Unfortunately, I'm not familiar with macros, I believe @timholy knows how to fix this.

Reply:
[Don't call `eval` from macros](https://stackoverflow.com/questions/52641915/calling-julia-functions-with-keyword-arguments-using-macro). You simply can't check types from a macro---they are lexical, meaning they operate on expressions not on objects.

The only thing you could do is to take every non-recognized symbol (e.g., not `:+`) and call `maybe_channelwise` on it and then have `maybe_channelwise` check the type of the object. But this will probably break fantastically for many things.